src/System.Net.HttpListener/tests/HttpListenerAuthenticationTests.cs

```diff
@@ -404,6 +424,24 @@ public void Realm_SetDisposed_ThrowsObjectDisposedException()
             return clientTask.Result;
         }
 
+        public async Task<HttpResponseMessage> AuthenticationFailureAsyncContext(HttpClient client, HttpStatusCode errorCode)
+        {
+            Task<HttpResponseMessage> clientTask = client.GetAsync(_factory.ListeningUrl);
+            Task<HttpListenerContext> serverTask = _listener.GetContextAsync();
+
+            // The client task should complete first - the server should send a 401 response.
+            Task resultTask = await Task.WhenAny(clientTask, serverTask);
+            if (resultTask == serverTask)
+            {
+                await serverTask;
```

Comment:
Why do we care about bubbling up exceptions from the server task when what we really want to do is have the Assert.Same hit below?